src/main/java/com/eafit/herzon/teis/models/User.java

```diff
@@ -0,0 +1,275 @@
+package com.eafit.herzon.teis.models;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreUpdate;
+import jakarta.persistence.Table;
+import jakarta.persistence.Temporal;
+import jakarta.persistence.TemporalType;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Represents a user entity in the system.
+ */
+@Entity
+@Table(name = "users")
+public class User {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  @Column(nullable = false, unique = true)
+  private String username;
+
+  @Column(nullable = false)
+  private String name;
+
+  @Column(nullable = false, unique = true)
+  private String email;
+
+  @Column(nullable = false)
+  private String address;
+
+  @Column(nullable = false)
+  private String password;
+
+  @Enumerated(EnumType.STRING)
+  @Column(nullable = false)
+  private Role role;
+
+  @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true)
```

Comment:
If you will retrieve creditCard from a user in code, you should change the fetch type to EAGER to solve the O(n+1) problem to a O(2). Watch example at [Auction Model](https://github.com/JuanFelipeRestrepoBuitrago/HerzonTEIS/blob/juan_felipe/dev/src/main/java/com/eafit/herzon/teis/models/Auction.java) or [Offer Model](https://github.com/JuanFelipeRestrepoBuitrago/HerzonTEIS/blob/juan_felipe/dev/src/main/java/com/eafit/herzon/teis/models/Offer.java) in my implementation
```suggestion
  @OneToMany(fetch = FetchType.EAGER, mappedBy = "user", 
      cascade = CascadeType.ALL, orphanRemoval = true)
  @JsonIgnore
  @Fetch(FetchMode.SUBSELECT)
```

Reply:
Fixed